samples/nrf9160/http_update/application_update/src/main.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <dfu/mcuboot.h>
+#include <modem/bsdlib.h>
+#include <net/fota_download.h>
+#include <stdio.h>
+#include "update.h"
+
+static void fota_dl_handler(const struct fota_download_evt *evt)
+{
+	switch (evt->id) {
+	case FOTA_DOWNLOAD_EVT_ERROR:
+		printk("Received error from fota_download\n");
+		/* Fallthrough */
+	case FOTA_DOWNLOAD_EVT_FINISHED:
+		update_sample_enable_button_irq();
+		break;
+
+	default:
+		break;
+	}
+}
+
+static void on_button_pressed(struct k_work *unused)
+{
+	int err;
+	char *apn = NULL;
+
+	/* Functions for getting the host and file */
+	err = fota_download_start(CONFIG_DOWNLOAD_HOST, CONFIG_DOWNLOAD_FILE,
+				  SEC_TAG, apn, 0);
+	if (err != 0) {
+		update_sample_enable_button_irq();
```

Comment:
Suggestion: Remove this function entirely, always reenable button IRQ, and just ignore `-EALREADY` errors from `fota_download_start`

Reply:
I'm not sure I understand what you need. I don't want to depend on the specific return values of some API as that is prone to change.

Are you suggesting to leave the button interrupt enabled at all time, and then simply disregard the callback when we have started a download?

Reply:
We seem to have different views on error codes :D. IMO they are there to be used, and I felt we could make things simpler. Also, `update_sample_enable_button_irq()` doesn't seem to match the rest of the API in terms of level of abstraction. Anyways, it's no big deal for me.

Reply:
I have broken so many APIs I will never trust one gain ;)